services/brig/src/Brig/User/Auth.hs

```diff
@@ -125,7 +122,33 @@ login (PasswordLogin li pw label _) typ = do
     AuthSuspended -> throwE LoginSuspended
     AuthEphemeral -> throwE LoginEphemeral
     AuthPendingInvitation -> throwE LoginPendingActivation
+  verifyCode code uid
   newAccess @ZAuth.User @ZAuth.Access uid typ label
+  where
+    verifyCode :: Maybe Code.Value -> UserId -> ExceptT LoginError (AppIO r) ()
+    verifyCode mbCode u = do
+      (mbEmail, mbTeamId) <- getEmailAndTeamId u
+      featureEnabled <- lift $ do
+        mbFeatureEnabled <- Intra.getVerificationCodeEnabled `traverse` mbTeamId
+        pure $ fromMaybe (Public.tfwoapsStatus Public.defaultTeamFeatureSndFactorPasswordChallengeStatus == Public.TeamFeatureEnabled) mbFeatureEnabled
+      when featureEnabled $ do
+        case mbCode of
+          Nothing -> loginFailedWith LoginCodeRequired u
+          Just c ->
+            case mbEmail of
+              Nothing -> loginFailedNoEmail u
+              Just email -> do
+                key <- Code.genKey <$> Code.mk6DigitGen (Code.ForEmail email)
```

Comment:
I think there is a refactoring hidden in here that makes it unnecessary to generate another code just to throw it away and keep the key, but I won't insist.  :)  perhaps add a FUTUREWORK?